src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/DefaultRazorSemanticTokenInfoService.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.AspNetCore.Razor.LanguageServer.Completion;
+using Microsoft.VisualStudio.Editor.Razor;
+using SyntaxNode = Microsoft.AspNetCore.Razor.Language.Syntax.SyntaxNode;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic
+{
+    internal class DefaultRazorSemanticTokenInfoService : RazorSemanticTokenInfoService
+    {
+        private readonly SemanticTokenLegend _symanticTokenLegend;
+
+        [ImportingConstructor]
+        public DefaultRazorSemanticTokenInfoService()
+        {
+            _symanticTokenLegend = new SemanticTokenLegend();
+        }
+
+        public override SemanticTokens GetSemanticTokens(RazorCodeDocument codeDocument, SourceLocation? location = null)
+        {
+            if (codeDocument is null)
+            {
+                throw new ArgumentNullException(nameof(codeDocument));
+            }
+            var syntaxTree = codeDocument.GetSyntaxTree();
+
+            var syntaxTokens = VisitAllNodes(syntaxTree);
+
+            var semanticTokens =  ConvertSyntaxTokensToSemanticTokens(syntaxTokens, codeDocument, _symanticTokenLegend);
+
+            return semanticTokens;
+        }
+
+        private static IEnumerable<SyntaxNode> VisitAllNodes(RazorSyntaxTree syntaxTree)
+        {
+            return VisitNode(syntaxTree.Root);
+        }
+
+        private static IReadOnlyList<SyntaxNode> VisitNode(SyntaxNode syntaxNode)
+        {
+            var result = new List<SyntaxNode>();
+
+            if (syntaxNode is null)
+            {
+                return result;
+            }
+
+            switch(syntaxNode.Kind)
+            {
+                case SyntaxKind.MarkupMinimizedTagHelperDirectiveAttribute:
+                    var minimized = (MarkupMinimizedTagHelperDirectiveAttributeSyntax)syntaxNode;
+                    result.Add(minimized.Name);
+                    break;
+                case SyntaxKind.MarkupTagHelperDirectiveAttribute:
+                    var directive = (MarkupTagHelperDirectiveAttributeSyntax)syntaxNode;
+                    result.Add(directive.Name);
```

Comment:
`MarkupTagHelperDirectiveAttributeSyntax.Name` would not include the `@`. Also, you'll want to include the directive attribute parameter if any `@onclick:preventDefault`.
So, you'll need to include all of the following,
- MarkupTagHelperDirectiveAttributeSyntax.Transition
- MarkupTagHelperDirectiveAttributeSyntax.Name
- MarkupTagHelperDirectiveAttributeSyntax.Colon
- MarkupTagHelperDirectiveAttributeSyntax.ParameterName

Reply:
Refer to this for the full syntax,
https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.AspNetCore.Razor.Language/Syntax/Syntax.xml#L212

Reply:
Actually, as per my conversation with @NTaylorMullen I think what I might do for this initial PR is to just not handle directives at all. They have a lot of edge cases and there's concern that this PR will be too large and hard to understand if I do it all here.

Reply:
Sounds good to me